README.md

```diff
@@ -90,43 +111,73 @@ worker.get('virtual/.', function(request, response) {
 });
 ```
 
-The output for any request that hits this url will be the original address of the url and a random number, with the headers that we specified in the response object.
+The output for any request that hits this URL will be the original address of the URL, and a randomly generated number. Note how we can even specify the headers for our response!
+
+Remember that the handler *must always return a Promise*.
+
+### onMessage(?) TODO
+### onBeforeEvicted(?) TODO
+### onEvicted(?) TODO
+
 
-One important point is the fact that when you write a handler for using in your request, the function handling those requests *must return a promise*.
+## Middleware examples
 
-## Can I see some examples of middleware?
+This package already incorporates two simple middlewares. They are available by default when you import it:
 
-Right we added 2 simple middlewares to this package, one for precaching static content and a simple offline cache handler. They are available by default when you import this library.
+TODO: I would suggest refactoring them away from this package
+
+### `StaticCacher`
+
+This will let you preload and cache static content during the `ServiceWorker` installation.
+
+TODO: where does `self` come from? in which context is this being executed?
```

Comment:
but if they are refactored to another module the example that uses `self` is invalid, right?